python_modules/libraries/dagster-dbt/dagster_dbt/dbt_project.py

```diff
@@ -106,33 +109,35 @@ def _should_use_built_artifacts(self) -> bool:
             bool(os.getenv("DAGSTER_DBT_PARSE_PROJECT_ON_LOAD"))
         )
 
-    @property
-    def _current_project_dir(self) -> Path:
-        if self._should_use_built_artifacts():
-            return self._packaged_project_dir if self._packaged_project_dir else self.project_dir
-
-        return self.project_dir
-
-    def _prepare_manifest(self, level: int = logging.DEBUG) -> Path:
+    @public
+    def create_manifest(self, log_level: int = logging.DEBUG) -> Path:
+        """Invoke `create_manifest_command` to create manifest.json."""
         logger.log(
-            level, f"Preparing dbt artifacts in {self.project_dir} at {self._target_folder}."
+            log_level,
+            f"Creating manifest.json for {self.project_dir} using {self._manifest_cmd}.",
         )
-        return (
+        manifest_path = (
             DbtCliResource(project_dir=os.fspath(self.project_dir))
             .cli(
-                self._prepare_command,
+                self._manifest_cmd,
                 target_path=self._target_folder,
             )
             .wait()
             .target_path.joinpath("manifest.json")
         )
+        logger.log(log_level, f"Created {manifest_path}.")
+        return manifest_path
 
-    def _handle_package_data(self, level: int) -> None:
+    @public
+    def sync_to_package_data(self, log_level: int) -> None:
+        """Sync `project_dir` directory to `packaged_project_dir` if `packaged_project_dir` is set."""
```

Comment:
Feels a little weird that this is a noop if the packaged_project_dir isn't set. Thoughts on raising here and doing the check inside `prepare_for_deployment`?